uaclient/apt.py

```diff
@@ -104,6 +147,18 @@ def run_apt_command(
     except util.ProcessExecutionError as e:
         if "Could not get lock /var/lib/dpkg/lock" in str(e.stderr):
             error_msg += " Another process is running APT."
+        else:
+            """
+            Treat errors where one of the APT repositories
+            is invalid or unreachable. In that situation, we alert
+            which repository is causing the error
+            """
+            rep_error = "The repository"
+            fetch_error = "Failed to fetch"
+            if rep_error in str(e.stderr) or fetch_error in str(e.stderr):
+                error_msg += "\n"
+                error_msg += _parse_apt_update_for_invalid_apt_config(e.stderr)
```

Comment:
I think _parse_apt_update_for_invalid_apt_config should prepend the "\n" itself. Also, since _parse_apt_update_for_invalid_apt_config does return "" if no match, we should be able to just call that directly.

```suggestion
    error_msg += _parse_apt_update_for_invalid_apt_config(e.stderr)
```

Reply:
The reason that I am appending here to `error_msg`  is that it already contains a error string that we return: 

```
APT update failed.
```

Maybe we can ignore this message and return the result directly. However, my only concern is the case that `_parse_apt_update_for_invalid_apt_config` returns empty. In that situation, we would not alert the user apt the apt update failure.

A second option is to pass the currently `error_msg` into the parse function and do the appending internally. What do you think @blackboxsw ?

Reply:
I still suggest we append, just let the _parse_apt_update function add the "\n" if it finds errors it can parse. In the event that it returns "" we don't end up altering error_msg because we error_msg +=  "".

Here's the full diff of whatI meant, _parse_apt_update already hasthe right logic in it for the most part (minus the leading "\n")

```diff
diff --git a/uaclient/apt.py b/uaclient/apt.py
index 20686b4..7d64a05 100644
--- a/uaclient/apt.py
+++ b/uaclient/apt.py
@@ -119,7 +119,7 @@ def _parse_apt_update_for_invalid_apt_config(apt_error: str) -> str:
                 failed_repos.add(repo_url_match)
 
     if failed_repos:
-        error_msg = status.MESSAGE_APT_UPDATE_INVALID_URL_CONFIG.format(
+        error_msg = "\n" + status.MESSAGE_APT_UPDATE_INVALID_URL_CONFIG.format(
             "s" if len(failed_repos) > 1 else "",
             "\n".join(sorted(failed_repos)),
         )
@@ -148,17 +148,8 @@ def run_apt_command(
         if "Could not get lock /var/lib/dpkg/lock" in str(e.stderr):
             error_msg += " Another process is running APT."
         else:
-            """
-            Treat errors where one of the APT repositories
-            is invalid or unreachable. In that situation, we alert
-            which repository is causing the error
-            """
-            rep_error = "The repository"
-            fetch_error = "Failed to fetch"
-            if rep_error in str(e.stderr) or fetch_error in str(e.stderr):
-                error_msg += "\n"
-                error_msg += _parse_apt_update_for_invalid_apt_config(e.stderr)
-
+            # Add any matching APT URL config error messages
+            error_msg += _parse_apt_update_for_invalid_apt_config(e.stderr)
         raise exceptions.UserFacingError(error_msg)
     return out
```